### DISCUSSION TITLE
postgres crashes and never recovers

### DISCUSSION THREAD

**Discussion by Cirion75** [immich-app/immich#6680] [Q&A] [answered]
Hello, I have been trying to test out Immich for a while now. And it starts great. but postgres just stops.
Import goes fine, but then postgres just stops.

What am I doing wrong, and how can I figure It out, nothing in the log makes sense.

I'm running Immich on Unraid 6.12.6
Immich:  ghcr.io/imagegenius/immich
Postgres:  tensorchord/pgvecto-rs:pg14-v0.1.11
Redis: redis

I also was able to set up NVIDIA transcoding with my RTX2060.

Entire library gets imported 1.3TB and transcoded and working great....
![image](https://github.com/immich-app/immich/assets/1172168/9754503b-14bf-4c3b-8431-b3bb7da895e7)

Then I suspect a night job does something, and Postgres gets a heart attack and shuts down.
[Postgreslog.txt](https://github.com/immich-app/immich/files/14071610/Postgreslog.txt)

If I restart Postgres, it just gives the same log.
If I delete all dockers and start all over, it works until the entire import is complete and until the first night...

I have tried googling several of the lines in the log, but no answer helps me.

I'm lost

**Comment by alextran1502** [comment]
cc @martabal any thoughts on this?

**Reply by mertalev** [reply]
I found another [post](https://forums.unraid.net/topic/149636-installing-extensions-on-postgresql-docker-container/?do=findComment&comment=1344289) in the Unraid forums about this from last month, also with the AIO image. 

The key lines are the ones that can't find postgres files: `FATAL:  could not open file "global/pg_filenode.map": No such file or directory`.

**Reply by martabal** [reply]
No idea, this is a postgres error and Unraid users use the same postgres image :thinking:

**Comment by mertalev** [comment]
Can you share the `immich_microservices` logs? Especially any from around the time this error happened.

**Reply by Cirion75** [reply]
There is no such file on my unraid. Where shoul I find it?

**Reply by mertalev** [reply]
Ah right, this is the AIO image. I'm interested in any job-related logs from around the time the crash happened, if there are any.

**Reply by Cirion75** [reply]
I started using this:
![image](https://github.com/immich-app/immich/assets/1172168/28264e3a-083d-4c5f-a4e6-3d3ad24caffb)

That makes an hourly backup of the Postgres database.
I did get one 23:06, 00:06 and 01:06 then nothing.
And the DB is crashed again.

Trying a restore now.

**Reply by Cirion75** [reply]
Is there a folder on the docker with those logs?

**Comment by bo0tzz** [comment]
Can you provide some detail about how you're running postgres and where its data directory is stored?

**Comment by Cirion75** [comment]
Just a docker on my Unraid.
Postgres: tensorchord/pgvecto-rs:pg14-v0.1.11

**Reply by bo0tzz** [reply]
"just a docker" is not enough information. Are you just using the standard community template? Are you storing the data in a docker volume or in a mounted folder? If the latter, where? etc.

**Reply by Cirion75** [reply]
Postgres is running from the standard comunity template, but since that is not working due to missing tensorchord I'm using the image from the Immich documentation: tensorchord/pgvecto-rs:pg14-v0.1.11
https://github.com/imagegenius/docker-immich#application-setup

It runs in a folder on Uraid. Path: /mnt/cache/appdata/postgresql14

Immich and Redis is the same, both from standard comunity templates. Running on the cache drive :
![image](https://github.com/immich-app/immich/assets/1172168/7fd254b0-cdd0-48fb-9aa3-f39e371f8250)
Free space on the SSD is 212GB
Free space on the Array is 24,8TB

Immich library is on the Array.
/mnt/user/Immich

**Reply by bo0tzz** [reply]
We've seen issues in the past with unraid cache disks where some nightly process comes along and moves a bunch of files off the cache, which will thoroughly break postgres and would explain the log lines you're seeing such as `FATAL:  could not open file "global/pg_filenode.map": No such file or directory`.

**Reply by Cirion75** [reply, marked as answer]
That actually makes a lot of sense.
![image](https://github.com/immich-app/immich/assets/1172168/922a31dc-dd88-4b73-b32f-16a603fc1e11)

My mover is set to run at 01:00 and after that Postgres panics and never recovers.
The files are still there, just not on the chache drive, and the path should not be cache.
As all new files and files worked on are on the cache and the mover then moves unused files to the Array.

That explains why the error just appears at night and not during the day for me.
This path is then wrong even if it was set by the community image;
/mnt/cache/appdata/postgresql14
Should be like this (like it is on my other dockers)
/mnt/user/appdata/postgresql14

I will test this and see.
